tests/integration/pilot/istioctl_test.go

```diff
@@ -172,3 +179,60 @@ func getPodID(i echo.Instance) (string, error) {
 
 	return "", fmt.Errorf("no workloads")
 }
+
+func TestAddToAndRemoveFromMesh(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		RunParallel(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "istioctl-add-to-mesh",
+				Inject: false,
+			})
+
+			var a echo.Instance
+
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&a, echo.Config{
+					Service:   "a",
+					Namespace: ns,
+					Ports: []echo.Port{
+						{
+							Name:     "http",
+							Protocol: protocol.HTTP,
+							// We use a port > 1024 to not require root
+							InstancePort: 8090,
+						},
+					},
+					Galley: g,
+					Pilot:  p,
+					Annotations: map[echo.Annotation]*echo.AnnotationValue{
+						echo.SidecarInject: {
+							Value: strconv.FormatBool(false)},
```

Comment:
I feel like its a bug that add-to-mesh ignores this annotation?

Reply:
no, it is not a bug of the add-to-mesh annotation.  It is our test framework where if this sidecar injection =false annotation isn't there, it will check for sidecar thus not return pod read.

Reply:
maybe I am reading this wrong, but it looks like you create a pod with the "do not inject" label and then run this command and it's injected? that doesn't seem right to me

Reply:
I don't think `remove-from-mesh` can remove injected sidecars yet (the work item is https://github.com/istio/istio/issues/20998 ).

Perhaps this test should focus on checking `add-to-mesh`.  The PR that improves remove-from-mesh can add an integration test for it.

Reply:
SO don't inject label only works on auto side car injection.   Basically, I'm saying don't inject so the test framework knows not to wait for sidecar.   Then use add-to-mesh to add it, then remove-from-mesh to remove it.  The key point I want to test is remove-from-mesh is clean and users can add-to-mesh again.   

Ed #20998 is for auto injection.  This integration test doesn't use auto injection.